### PR TITLE
Test ipa unique

### LINKED ISSUE #2969: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid ...
As seen here: http://files.cockpit-project.org/hubbot/744bb9246fc48e5bfc6231f9e72e5c7ed627ce7c_r7_x86-64/hubbot.html

```
Password for admin@COCKPIT.LAN: 
Journal database copied to check-realms-testIpa-10.111.112.177-FAIL.journal
.libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
libvirt: Domain Config error : operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
EWrote check-realms-testIpa-FAIL.png
Warning: Permanently added '10.111.112.177' (ECDSA) to the list of known hosts.

======================================================================
ERROR: testIpa (__main__.TestRealms)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-realms", line 40, in testIpa
    ipa.start()
  File "/home/hubbot/hubbot/744bb9246fc48e5bfc6231f9e72e5c7ed627ce7c_r7_x86-64/test/testvm.py", line 856, in start
    self._start_qemu(maintain=maintain, macaddr=macaddr)
  File "/home/hubbot/hubbot/744bb9246fc48e5bfc6231f9e72e5c7ed627ce7c_r7_x86-64/test/testvm.py", line 814, in _start_qemu
    dom = self.virt_connection.createXML(test_domain_desc, libvirt.VIR_DOMAIN_START_AUTODESTROY)
  File "/usr/lib64/python2.7/site-packages/libvirt.py", line 3427, in createXML
    if ret is None:raise libvirtError('virDomainCreateXML() failed', conn=self)
libvirtError: operation failed: domain 'ipa-fedora-22-x86_64-3_f0.cockpit.lan' already exists with uuid e4034777-1b0b-4e68-acbd-eebd6cdde165
```

### PR DESCRIPTION
**Pull request by dperpeet**
this should fix #2969

### CODE REVIEW

**Comment by stefwalter** [comment]
This had the same failure as #2969 in its r7 test: http://files.cockpit-project.org/hubbot/932d6d648dafa8e835282d8d847ec29ba6b42587_r7_x86-64/hubbot.html

File: test/check-realms [hunk 1]
```diff
@@ -112,6 +112,10 @@ class TestRealms(MachineCase):
         b.click(".realms-op-cancel")
         b.wait_popdown("realms-op")
 
+        # free the ipa domain
```
Comment (mvollmer) [inline test/check-realms]: The term "domain" is confusing here, since it might also refer to the FreeIPA concept.

Comment (dperpeet) [inline test/check-realms]: > The term "domain" is confusing here, since it might also refer to the FreeIPA concept.

good point! fixed.

**Comment by dperpeet** [comment]
libvirt by default reserves the right to gracefully terminate a virtual machine / domain - this can lead to delays
I changed the code so that `testvm.kill()` explicitly uses the non-graceful libvirt way of destroying a virtual machine.

**Pull request closed without merging**